packages/abi-coder/src/coders/b512.ts

```diff
@@ -0,0 +1,35 @@
+import { arrayify } from '@ethersproject/bytes';
+import { bn, toHex } from '@fuel-ts/math';
+
+import Coder from './abstract-coder';
+
+export default class B512Coder extends Coder<string, string> {
```

Comment:
If I understand it correctly, the tests for all coders are condensed in a single<sup>(*)</sup> file, right? Please add a new one for `b512`.
 - `src/coders/coders.test.ts`

> _<sup>(*)</sup> Although I find it very confusing to have tests selectively grouped in any way that's not side-by-side with their source counterpart, this is how it seems to have been done in this case._